lib/dialects/mysql/query.js

```diff
@@ -92,7 +92,22 @@ class Query extends AbstractQuery {
       this.handleInsertQuery(data);
 
       if (!this.instance) {
-        result = data[this.getInsertIdField()];
+        // handle bulkCreate AI primiary key 
+        if (
+          data.constructor.name === 'ResultSetHeader'
+          && this.model
+          && this.model.autoIncrementField
+          && this.model.autoIncrementField === this.model.primaryKeyAttribute
+          && this.model.rawAttributes[this.model.primaryKeyAttribute]
+        ) {
+          const startId = data[this.getInsertIdField()];
+          result = [];
+          for (let i = startId; i < startId + data.affectedRows; i++) {
```

Comment:
Non-1 increment is rare i suppose. InnoDB is usually the storage engine of choice.
What about SQLite? And how do we test that this holds up under concurrency (i'd imagine it does though)